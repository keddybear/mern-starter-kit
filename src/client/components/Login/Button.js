import React from 'react';
import { shallowEqual } from 'helpers/shallowEqual';

class Button extends React.Component {
	shouldComponentUpdate(nextProps) {
		return !shallowEqual(this.props, nextProps);
	}

	render() {
		const { isLogin, showLogin, onClick } = this.props;
		let button;
		let name;

		if (isLogin) {
			name = 'Log In';
		} else {
			name = 'Sign Up';
		}

		if ((showLogin === 1 && isLogin) || (showLogin === 2 && !isLogin)) {
			button = (
				<div className='de-btn-active'>
					<span>{name}</span>
				</div>
			);
		} else {
			button = (
				<div className='de-btn' onClick={onClick} onKeyPress={() => {}} tabIndex='-1' role='button'>
					<span>{name}</span>
				</div>
			);
		}

		return (
			<div>
				{button}
			</div>
		);
	}
}

export default Button;
