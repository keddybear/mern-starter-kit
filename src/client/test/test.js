/* eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

import Button from 'components/Login/Button';

// Test Button after refactoring Login/Signup Button into one
it('Toggle Login/Signup Button', () => {
	const LoginButtonActive = renderer.create(
		<Button isLogin showLogin={1} />
	);

	const LoginButtonInactive = renderer.create(
		<Button isLogin showLogin={2} />
	);

	const LoginButtonInactive2 = renderer.create(
		<Button isLogin showLogin={3} />
	);

	const SignupButtonActive = renderer.create(
		<Button isLogin={false} showLogin={2} />
	);

	const SignupButtonInactive = renderer.create(
		<Button isLogin={false} showLogin={1} />
	);

	const SignupButtonInactive2 = renderer.create(
		<Button isLogin={false} showLogin={3} />
	);

	expect(LoginButtonActive.toJSON()).toMatchSnapshot();
	expect(LoginButtonInactive.toJSON()).toMatchSnapshot();
	expect(LoginButtonInactive2.toJSON()).toMatchSnapshot();
	expect(SignupButtonActive.toJSON()).toMatchSnapshot();
	expect(SignupButtonInactive.toJSON()).toMatchSnapshot();
	expect(SignupButtonInactive2.toJSON()).toMatchSnapshot();
});
