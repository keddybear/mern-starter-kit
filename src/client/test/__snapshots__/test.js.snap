// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle Login/Signup Button 1`] = `
<div>
  <div
    className="de-btn-active"
  >
    <span>
      Log In
    </span>
  </div>
</div>
`;

exports[`Toggle Login/Signup Button 2`] = `
<div>
  <div
    className="de-btn"
    onKeyPress={[Function]}
    role="button"
    tabIndex="-1"
  >
    <span>
      Log In
    </span>
  </div>
</div>
`;

exports[`Toggle Login/Signup Button 3`] = `
<div>
  <div
    className="de-btn"
    onKeyPress={[Function]}
    role="button"
    tabIndex="-1"
  >
    <span>
      Log In
    </span>
  </div>
</div>
`;

exports[`Toggle Login/Signup Button 4`] = `
<div>
  <div
    className="de-btn-active"
  >
    <span>
      Sign Up
    </span>
  </div>
</div>
`;

exports[`Toggle Login/Signup Button 5`] = `
<div>
  <div
    className="de-btn"
    onKeyPress={[Function]}
    role="button"
    tabIndex="-1"
  >
    <span>
      Sign Up
    </span>
  </div>
</div>
`;

exports[`Toggle Login/Signup Button 6`] = `
<div>
  <div
    className="de-btn"
    onKeyPress={[Function]}
    role="button"
    tabIndex="-1"
  >
    <span>
      Sign Up
    </span>
  </div>
</div>
`;
